mne/decoding/tests/test_time_gen.py

```diff
@@ -171,6 +171,25 @@ def test_generalization_across_time():
     assert_true(len(gat.test_times_['slices'][0]) == 15 ==
                 np.shape(gat.scores_)[1])
 
+    # Test score_mode
+    gat.score_mode = 'foo'
+    assert_raises(ValueError, gat.score, epochs)
+    gat.score_mode = 'fold-wise'
+    scores = gat.score(epochs)
+    assert_array_equal(np.shape(scores), [15, 15, 5])
+    gat.score_mode = 'sample-wise'
+    scores = gat.score(epochs)
+    assert_array_equal(np.shape(scores), [15, 15])
+    gat.score_mode = 'mean-fold-wise'
+    scores = gat.score(epochs)
+    assert_array_equal(np.shape(scores), [15, 15])
+    gat.predict_mode = 'mean-prediction'
+    with warnings.catch_warnings(record=True) as w:
+        gat.score(epochs)
+        assert_true(len(w) > 0)
```

Comment:
Guys do we need some concrete usage oriented checking of the code as it
currently is? I would be happy schedule some time to do that over the next
2 days.

On Sat, Apr 30, 2016 at 8:07 PM Eric Larson <notifications@github.com>
wrote:

> In mne/decoding/tests/test_time_gen.py
> <https://github.com/mne-tools/mne-python/pull/3143#discussion_r61672809>:
>
> > +    # Test score_mode
> > +    gat.score_mode = 'foo'
> > +    assert_raises(ValueError, gat.score, epochs)
> > +    gat.score_mode = 'fold-wise'
> > +    scores = gat.score(epochs)
> > +    assert_array_equal(np.shape(scores), [15, 15, 5])
> > +    gat.score_mode = 'sample-wise'
> > +    scores = gat.score(epochs)
> > +    assert_array_equal(np.shape(scores), [15, 15])
> > +    gat.score_mode = 'mean-fold-wise'
> > +    scores = gat.score(epochs)
> > +    assert_array_equal(np.shape(scores), [15, 15])
> > +    gat.predict_mode = 'mean-prediction'
> > +    with warnings.catch_warnings(record=True) as w:
> > +        gat.score(epochs)
> > +        assert_true(len(w) > 0)
>
> Do you have repo notifications on? That's a good way to keep track of my
> errors :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3143/files/efca4fae9ae2a0f48e8a188267d48a0a7cb9346a#r61672809>
>